### PR TITLE
Add cpp_info.name to cmake and pkg_config generators

### PR DESCRIPTION
**Pull request by danimtb**
Changelog: Feature: Use `cpp_info.name` in CMake and pkg_config generators.
Docs: https://github.com/conan-io/docs/pull/1393

@tags: slow

- [x] Refer to the issue that supports this Pull Request: related to #5090 
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [x] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

**Comment by danimtb** [comment]
That way the name of dependencies in the dict for `deps_cpp_info` would not match the name of the references and it would be confusing. The name of the reference should not change IMO

**Comment by lasote** [comment]
Maybe that is exactly what we want as far as we document it properly. It looks like it simplifies everything A LOT.

File: conans/test/unittests/client/generators/cmake_test.py [hunk 1]
```diff
@@ -51,10 +52,12 @@ def variables_setup_test(self):
         conanfile.initialize(Settings({}), EnvValues())
         ref = ConanFileReference.loads("MyPkg/0.1@lasote/stables")
         cpp_info = CppInfo("dummy_root_folder1")
+        cpp_info.name = ref.name
```
Comment (lasote) [inline conans/test/unittests/client/generators/cmake_test.py]: What happens if you don't assign this?

Comment (danimtb) [reply conans/test/unittests/client/generators/cmake_test.py]: When `cpp_info.name` is used inside the generator, the None type makes it crash

`AttributeError: 'NoneType' object has no attribute 'upper'`

**Comment by danimtb** [comment]
Done!

File: conans/test/unittests/client/generators/pkg_config_test.py [hunk 2]
```diff
@@ -16,6 +16,7 @@ def variables_setup_test(self):
         conanfile.initialize(Settings({}), EnvValues())
         ref = ConanFileReference.loads("MyPkg/0.1@lasote/stables")
         cpp_info = CppInfo("dummy_root_folder1")
+        cpp_info.name = ref.name
```
Comment (lasote) [inline conans/test/unittests/client/generators/pkg_config_test.py]: Shouldn't you try to change the name and test something?

Comment (danimtb) [reply conans/test/unittests/client/generators/pkg_config_test.py]: Sure, done!

**Pull request merged**